torch/csrc/jit/codegen/cuda/arith.cpp

```diff
@@ -686,13 +686,25 @@ TensorView* arange(Val* start, Val* end, DataType dtype) {
 
 TensorView* arange(Val* start, Val* end, Val* step, DataType dtype) {
   if (isIntegralType(dtype)) {
-    start = castOp(DataType::Int, start);
-    end = castOp(DataType::Int, end);
-    step = castOp(DataType::Int, step);
+    if (start->getDataType() != DataType::Int) {
+      start = castOp(DataType::Int, start);
+    }
+    if (end->getDataType() != DataType::Int) {
+      end = castOp(DataType::Int, end);
+    }
+    if (step->getDataType() != DataType::Int) {
+      step = castOp(DataType::Int, step);
+    }
   } else if (isFloatingPointType(dtype)) {
-    start = castOp(DataType::Double, start);
-    end = castOp(DataType::Double, end);
-    step = castOp(DataType::Double, step);
+    if (start->getDataType() != DataType::Double) {
+      start = castOp(DataType::Double, start);
+    }
+    if (end->getDataType() != DataType::Double) {
+      end = castOp(DataType::Double, end);
+    }
+    if (step->getDataType() != DataType::Double) {
+      step = castOp(DataType::Double, step);
+    }
```

Comment:
This is just to make the generated code cleaner, because I don't like a lot of things like `i5 = i1`.